syntax = "proto3";

package fixbug;

option cc_generic_services = true;

message ResultCode 
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

// 两个RPC调用的请求和响应分别定义：
// Login()的请求和响应分别为LoginRequest  LoginResponse
// Register()的请求和响应分别为 RegisterRequest RegisterResponse
// 然后这两个RPC调用都属于UserServiceRpc服务。
message LoginRequest 
{
    bytes name = 1;
    bytes pwd = 2;
}

message LoginResponse
{
    ResultCode result = 1;
    bool success = 2;
}

message RegisterRequest
{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

message RegisterResponse
{
    ResultCode result = 1;
    bool success = 2;
}

service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
}

